apps/keycloak.py

```diff
@@ -0,0 +1,379 @@
+import json
+
+from transpire import helm
+from transpire.resources import ConfigMap
+from transpire.utils import get_versions
+
+name = "keycloak"
+
+
+def oidc_client(
+    client_id: str,
+    *,
+    name: str,
+    description: str,
+    home_url: str,
+    redirect_uris: list[str],
+):
+    return {
+        "clientId": client_id,
+        "name": name,
+        "description": description,
+        "baseUrl": home_url,
+        "clientAuthenticatorType": "client-secret",
+        "redirectUris": redirect_uris,
+        "webOrigins": ["+"],
+        "publicClient": False,
+        "protocol": "openid-connect",
+        "enabled": True,
+    }
+
+
+clients = [
+    oidc_client(
+        "argocd",
+        name="ArgoCD",
+        description="Declarative GitOps CD for Kubernetes",
+        home_url="https://argo.ocf.berkeley.edu/",
+        redirect_uris=["https://argo.ocf.berkeley.edu/auth/callback"],
+    ),
+    oidc_client(
+        "hedgedoc",
+        name="HedgeDoc",
+        description="Collaborative Markdown Notes",
+        home_url="https://notes.ocf.berkeley.edu/",
+        redirect_uris=["https://notes.ocf.berkeley.edu/auth/oauth2/callback"],
+    ),
+]
+
+keycloak_config_cli = {
+    "id": "ocf",
+    "realm": "ocf",
+    "displayName": "OCF",
+    "sslRequired": "all",
+    "enabled": True,
+    "registrationAllowed": False,
+    "loginWithEmailAllowed": False,
+    "loginTheme": "ocf-theme",
```

Comment:
I don't see the theme in this PR, where is it stored? (I think this won't work!)

I think you should use this theme, it's probably the nicest: https://github.com/lukin/keywind

Reply:
isn't added yet

Reply:
keywind added